core/application/reconcile.go

```diff
@@ -165,6 +169,13 @@ func (a *app) reconcile(ctx context.Context, spec *models.MicroVM, logger *logru
 	spec.Status.Retry = 0
 	spec.Status.NotBefore = 0
 
+	// I know this check is somewhat redundant since we would have exited above if
+	// it was Delete instead, but leaving in just in case the above gets removed
+	// at some point.
+	if plan.Name() == plans.MicroVMCreateOrUpdatePlanName {
+		spec.Status.State = models.CreatedState
+	}
```

Comment:
Can this be part of the CreateOrUpdate plan? Like a Finalize/PostHook/PostSomething function on plans? It feels like, if we add small things in the reconcile function because it's "just a small change" we end up with a lot of extra in the function that can be handled somewhere else.

Reply:
🤔  yeh i will think of a nice way to do that.

ideally we would shove all status-setting bits down into plans in the form of a hook like you say; right now there are lots of little status alterations happening to hat pointer both in reconcile and in plan which makes it quite hard to track.